src/Umbraco.Web/Mvc/RenderRouteHandler.cs

```diff
@@ -279,10 +280,40 @@ internal virtual RouteDefinition GetUmbracoRouteDefinition(RequestContext reques
                     //set the controller and name to the custom one
                     def.ControllerType = controllerType;
                     def.ControllerName = ControllerExtensions.GetControllerName(controllerType);
+
                     if (def.ControllerName != defaultControllerName)
                     {
                         def.HasHijackedRoute = true;
                     }
+
+                    // Look for hijacked action
+                    var hijackActions = controllerType
```

Comment:
We will need to perf optimize this before release we don't want all this reflection happening on each request. 
//cc @nul800sebastiaan + @clausjensen - we can add a task for this to a sprint.